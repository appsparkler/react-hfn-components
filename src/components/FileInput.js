import React from 'react'

const DEFAULT_CONFIG = {
  maxBytes: 5000,
}

export default React.forwardRef(function({config=DEFAULT_CONFIG}, ref) {
  return (
    <input
      type="file"
      ref={ref}
      className={ config.classNames }
      onInput={ setFilesToUpload.bind(null, config) }
      onChange={ () => console.log('changed...')}
      multiple={config.multiple}
    />
  )
})

function setFilesToUpload({
  maxByteExceededCallback,
  maxBytes,
  setFilesToUpload,
}, evt) {
  try {
    const {target} = evt
    const filesToUpload = [...target.files]
    const totalBytes = filesToUpload.reduce((r, file) => r + file.size, 0)
    if (totalBytes > maxBytes) {
      maxByteExceededCallback.call(null, arguments)
      target.value = null
      setFilesToUpload([])
    } else {
      setFilesToUpload([
        ...target.files,
      ])
    }
  } catch (err) {
    console.error(err)
  }
}
