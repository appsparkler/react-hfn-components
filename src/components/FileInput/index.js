import React from 'react'
import useFileInput from './useFileInput'
import ProgressBar from '@react-hfn-components/ProgressBar'

function FileInput(props) {
  const {
    handleInput,
    validationError,
    maxBytes,
    type,
    uploadDetail,
    isUploading,
    file,
  } = useFileInput({props})
  return (
    <div className={ `FileInput ${file && 'edit-file'}`}>
      {isUploading && <p>Uploading...</p>}
      {!isUploading && (
        <label >
          { props && props.label }
          <br />

          <input
            type={type}
            disabled={props.disabled || false}
            name={props.name || 'file'}
            required={props.required || false}
            onInput={handleInput}
          />
        </label>
      )}
      {
        uploadDetail && (
          <ProgressBar
            key={uploadDetail.key}
            uploadDetail={uploadDetail}
          />
        )
      }
      <p className="info">
        {(props && props.maxBytesInfoMessage) || 'Max file size: '}
        {(maxBytes/(1024 * 1024)).toFixed(2)} MB
      </p>
      <p className="validation-error">{validationError}</p>
    </div>
  )
}

export default FileInput
