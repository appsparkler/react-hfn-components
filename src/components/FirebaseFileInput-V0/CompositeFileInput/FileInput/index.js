import React from 'react'
import useFileInput from './useFileInput'
import ProgressBar from './ProgressBar'
import HelpBlock from '@react-hfn-components/HelpBlock'
import BSAlert from '@react-hfn-components/BSAlert'

function FileInput(props) {
  debugger
  const {
    handleInput,
    validationError,
    maxBytes,
    type,
    uploadDetail,
    isUploading,
    file,
  } = useFileInput({props})
  return (
    <div className={ `FileInput ${file && 'edit-file'}`}>
      {!isUploading && (
        <div className="form-group">
          <label>
            <span className="label-text">{ props && props.label }</span>
            <span className="required-asterix">{ props?.required && '*'}</span>
          </label>
          {props?.components?.fileInput && (
            <props.components.fileInput
              type={type}
              disabled={props.disabled || false}
              name={props.name || 'file'}
              required={props.required || false}
              onInput={handleInput}
            />) || (<input
              type={type}
              disabled={props.disabled || false}
              name={props.name || 'file'}
              required={props.required || false}
              onInput={handleInput}
            />)}

        </div>
      )}
      {
        uploadDetail && isUploading && (
          <ProgressBar
            key={uploadDetail.key}
            uploadDetail={uploadDetail}
          />
        )
      }
      {isUploading && <HelpBlock msg={ file ? 'Editing...': 'Uploading...'} />}
      <HelpBlock msg={
        `${(props && props.maxBytesInfoMessage) || 'Max file size: '}
         ${(maxBytes/(1024 * 1024)).toFixed(2)} MB`
      }
      />
      {validationError && <BSAlert type="danger" msg={validationError} />}
    </div>
  )
}

export default FileInput
