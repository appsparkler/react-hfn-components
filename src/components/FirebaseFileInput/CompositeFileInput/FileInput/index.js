import React from 'react'
import HelpBlock from '@react-hfn-components/HelpBlock'
import BSAlert from '@react-hfn-components/BSAlert'
import BSProgress from '@react-hfn-components/BSProgress'
import FirebaseFileLink from '@react-hfn-components/FirebaseFileLink'
import PropTypes from 'prop-types'
import useFileInput from './useFileInput'

const FileInput = (props) => {
  const {
    components, type, name, label, maxBytes, maxBytesError, progress,
    handleInput, maxBytesExceeded, isUploading, file,
  } = useFileInput(props)
  return (
    <>
      {components?.fileInput && (
        <components.fileInput
          type={type}
          name={name}
          label={label}
          maxBytes={maxBytes}
          maxBytesError={maxBytesError}
          progress={progress}
          handleInput={handleInput}
          maxBytesExceeded={maxBytesExceeded}
          isUploading={isUploading}
          file={file}
        />
      )}

      {!components?.fileInput && (
        <div className="form-group">
          <FirebaseFileLink storageRef={props.storageRef} />
          <label>{label}</label>
          <input
            type={type}
            name={name}
            onInput={handleInput}
          />
          <br />
          <BSProgress
            type="info"
            striped="true"
            progress="30"
          />
          <HelpBlock msg={`${maxBytes.toFixed()}MB`} />
          {maxBytesExceeded && (
            <BSAlert type="danger" msg={`${maxBytesError}`} />
          )}
        </div>
      )}
    </>
  )
}

FileInput.propTypes = {
  components: PropTypes.object,

  label: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,

  maxBytes: PropTypes.number,
  maxBytesError: PropTypes.string,

  storageRef: PropTypes.object,
}

FileInput.defaultProps = {
  maxBytes: 1 * 1024 * 1024, // 1MB
  name: 'file',
  type: 'file',
}

export default FileInput
