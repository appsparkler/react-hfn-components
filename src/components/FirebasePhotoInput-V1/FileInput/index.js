import React from 'react'
import PropTypes from 'prop-types'
import useFileInput from './useFileInput'

const FileInput = (props) => {
  const {handleChange} = useFileInput({props})
  return (
    <div className="FileInput">
      <div className="form-group">
        <label>
          <span className="label-text">Upload Photo</span>
          <span className="required-asterix"> *</span>
          <br />
          <input
            type="file"
            className="form-control"
            onChange={ handleChange }
          />
        </label>
      </div>
    </div>
  )
}

FileInput.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.object,
  onUpload: PropTypes.func,
  storageRef: PropTypes.object,
  config: PropTypes.object,
  // required: PropTypes.boolean,
  // disabled: PropTypes.boolean,
  // readOnly: PropTypes.boolean,
  // noLabel: PropTypes.boolean,
}

export default FileInput
