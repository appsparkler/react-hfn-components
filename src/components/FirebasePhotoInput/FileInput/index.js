import React from 'react'
import PropTypes from 'prop-types'
import useFileInput from './useFileInput'

const FileInput = (props) => {
  const {handleChange} = useFileInput({props})
  return (
    <div className="FileInput">
      <label>
        <span className="label-text">Upload Photo</span>
        <span className="required-asterix"> *</span>
        <br />
        <input
          type="file"
          onChange={ handleChange }
        />
      </label>
    </div>
  )
}

FileInput.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.object,
  onUpload: PropTypes.object,
  storageRef: PropTypes.object,
  config: PropTypes.object,
  // required: PropTypes.boolean,
  // disabled: PropTypes.boolean,
  // readOnly: PropTypes.boolean,
  // noLabel: PropTypes.boolean,
}

export default FileInput
