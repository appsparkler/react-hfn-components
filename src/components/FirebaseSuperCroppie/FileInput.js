import React from 'react'
import PropTypes from 'prop-types'
import {
  Stack,
} from 'office-ui-fabric-react'
import FileInputButton from '@react-hfn-hoc/FileInputButton'

const FileInput = ({
  file, fileInputRef, handleFileInputChange,
  isUploading, isVerifying, imgIsLoading,
  capture, accept,
}) => (
  <Stack.Item>
    <div style={{width: '100vw'}}>
      <Stack horizontalAlign="center">
        <Stack.Item>
          <FileInputButton
            primary={!!file}
            secondary={!file}
            disabled={isUploading || isVerifying || imgIsLoading}
            onChange={handleFileInputChange}
            text={file ? 'Edit Photo': 'Upload Photo'}
            fileInputRef={fileInputRef}
            capture={capture}
            accept={accept}
          />
        </Stack.Item>
      </Stack>
    </div>
  </Stack.Item>
)

FileInput.propTypes = {
  file: PropTypes.object,
  fileInputRef: PropTypes.shape({
    current: PropTypes.any,
  }),
  handleFileInputChange: PropTypes.func.isRequired,
  isUploading: PropTypes.bool.isRequired,
  isVerifying: PropTypes.bool.isRequired,
  imgIsLoading: PropTypes.bool.isRequired,
  capture: PropTypes.string,
  accept: PropTypes.string,
}

export default FileInput
