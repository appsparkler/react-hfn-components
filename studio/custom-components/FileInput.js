import React from 'react'
import PropTypes from 'prop-types'

const FileInput = (props) => {
  const {handleChange} = props
  return (
    <div className="FileInput">
      <div className="form-group">
        <label>
          <span className="label-text">Upload Photo</span>
          <span className="required-asterix"> *</span>
          <br />
          <input
            type="file"
            className="form-control"
            onChange={ handleChange }
          />
        </label>
      </div>
    </div>
  )
}

FileInput.propTypes = {
  handleChange: PropTypes.func,
  // name: PropTypes.string,
  // label: PropTypes.string,
  // value: PropTypes.object,
  // onUpload: PropTypes.func,
  // storageRef: PropTypes.object,
  // config: PropTypes.object,
}

export default FileInput
