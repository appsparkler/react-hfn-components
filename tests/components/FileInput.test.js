import React from 'react'
import {shallow, configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import FileInput from '../../src/components/FileInput'
import {
  setFilesToUpload as setFilesToUpload2,
} from '../../src/components/FileInput'

configure({adapter: new Adapter()})

describe('setFilesToUpload function', () => {
  it(`should the method config.setFilesToUpload
    to have been called correctly`, () => {
    const [filesToUpload, setFilesToUpload] = [[], jest.fn()]
    const fileInputRef = {}
    const config = {
      filesToUpload,
      setFilesToUpload,
    }
    const evt = {
      target: {
        files: [
          new File(['foo'], 'file-1.txt'),
          new File(['bar'], 'file-2.txt'),
        ],
      },
    }
    setFilesToUpload2(config, fileInputRef, evt)
    expect(config.setFilesToUpload).toHaveBeenCalledTimes(1)
  })
})

describe('FileInput Rendering', () => {
  it('should correctly render FileInput component', () => {
    const wrapper = shallow(<FileInput />)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
