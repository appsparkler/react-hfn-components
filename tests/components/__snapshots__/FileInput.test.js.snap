// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput Rendering should correctly render FileInput component 1`] = `"<input type=\\"file\\" class=\\"\\"/>"`;
